pkgs/os-specific/linux/disk-indicator/default.nix

```diff
@@ -1,7 +1,7 @@
 { stdenv, fetchgit, libX11 }:
 
 stdenv.mkDerivation {
-  name = "disk-indicator";
+  name = "disk-indicator-git-2014-05-19";
```

Comment:
Perhaps `nix-env` should just expect a more explicit version field, rather than trying to parse a package name, but that's a much bigger change.